framework/util/page_guard_manager_uffd.cpp

```diff
@@ -215,7 +215,8 @@ bool PageGuardManager::UffdSetSignalHandler()
     // Install signal handler for the RT signal
     {
         struct sigaction sa = {};
-        sa.sa_flags         = SA_SIGINFO;
+        // Minimize side effects of the RT signal handler by restarting interrupted system calls when possible
+        sa.sa_flags = SA_RESTART;
         sigemptyset(&sa.sa_mask);
         sa.sa_handler = UffdStaticSignalHandler;
```

Comment:
I was about to comment that since `SA_SIGINFO` was removed the signal handler should be registered in `sa.sa_sigaction` instead of `sa.sa_handler` but it seems it's the other way around. So it was wrong all the time but now should be correct.